### PR TITLE
gtk: Add run_future() for native dialogs

### PR DESCRIPTION
**Pull request by sophie-h**
Port gtk-rs/gtk-rs#500

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
hmm, seems like I should drop the x11/wayland features I guess

File: gtk4/src/native_dialog.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::prelude::*;
+use crate::NativeDialog;
+use crate::ResponseType;
+use std::cell::Cell;
+use std::future::Future;
+use std::pin::Pin;
+
+pub trait NativeDialogExtManual {
```
Comment (bilelmoussaoui) [inline gtk4/src/native_dialog.rs:10]: should be added to prelude

Comment (bilelmoussaoui) [reply gtk4/src/native_dialog.rs:10]: oups, never mind

File: gtk4/src/native_dialog.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // rustdoc-stripper-ignore-next
+    /// Shows the dialog and returns a `Future` that resolves to the
+    /// `ResponseType` on response.
+    ///
+    /// ```no_run
+    /// # use gtk4 as gtk;
```
Comment (bilelmoussaoui) [inline gtk4/src/native_dialog.rs:16]:
```suggestion
```
there's no mention of gtk4 in the docs, it's mostly a crate name

**Review by sdroege — APPROVED** [review]

File: gtk4/src/native_dialog.rs (hunk 2 above)
Comment (sophie-h) [reply gtk4/src/native_dialog.rs:16]: What do you mean? I think we need this line for the example to compile.

Comment (bilelmoussaoui) [reply gtk4/src/native_dialog.rs:16]: We already assume the package is renamed from gtk4 to gtk.

Comment (sophie-h) [reply gtk4/src/native_dialog.rs]: Added your change, no the tests fail because the example cannot compile.

Comment (bilelmoussaoui) [reply gtk4/src/native_dialog.rs]: wait, was that part of the code and not just a comment?

Comment (sophie-h) [reply gtk4/src/native_dialog.rs]: Yes. The hash just hides it in the docs:

https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example

Comment (bilelmoussaoui) [reply gtk4/src/native_dialog.rs]: aah oh I see, TIL. My bad then, please add it back :)

Comment (sophie-h) [reply gtk4/src/native_dialog.rs:16]: Done :)

**Comment by bilelmoussaoui** [comment]
Thanks Sophie! :)

**Pull request merged**